### DISCUSSION TITLE
Issues renaming column to contain UK pound sign £

### DISCUSSION THREAD

**Discussion by davb06** [johnkerl/miller#784] [Q&A] [answered]
I am having problems when renaming a column to "Employer contribution (£)", but the new column name appears as "Employer contribution (┬ú)". Here is a hex view of the output:
![mlr_hex_output](https://user-images.githubusercontent.com/59541218/146422697-d01013c1-e27a-4854-8064-b96a6ccc171a.jpg)
The complete command I used is:
`mlr --csv cut -f name1,surname,alloc_nino,ref,sgp,eepens,erpens then put -S '$Title="MS";${Date of birth}=""' then reorder -f Title,name1,surname,"Date of birth",alloc_nino,ref,sgp,eepens,erpens then rename name1,"First name",surname,"Last name",alloc_nino,"National Insurance number",ref,"Payroll reference",sgp,"Earnings in contribution period",eepens,"Worker contribution (£)",erpens,"Employer contribution (£)" "%inputfile%" >"%outputfile%"`
(The file names are set in the batch file.)
Input data used: [inputtest.CSV](https://github.com/johnkerl/miller/files/7729565/inputtest.CSV)

Can I modify my command - or is this a bug?

Curiously, this problem does not arise with a much more limited command, where I cut and rename a single column:
`mlr --csv cut -f erpens then rename erpens,"Employer contribution (£)" input.csv >output.csv`
```
erpens
5.00
```
This mini input file: [input.CSV](https://github.com/johnkerl/miller/files/7728588/input.CSV)

Am running on Windows with the latest v6 beta build.

(I have updated my post, to clarify my real problem.)

**Comment by aborruso** [comment]
Hi,
If I run in win

```
mlr.exe --csv cut -f name1,surname,alloc_nino,ref,sgp,eepens,erpens then put -S '$Title="MS";${Date of birth}=""' then reorder -f Title,name1,surname,"Date of birth",alloc_nino,ref,sgp,eepens,erpens then rename name1,"First name",surname,"Last name",alloc_nino,"National Insurance number",ref,"Payroll reference",sgp,"Earnings in contribution period",eepens,"Worker contribution (£)",erpens,"Employer contribution (£)" inputtest.CSV
```

I have

```
+-------+------------+-----------+---------------+---------------------------+-------------------+---------------------------------+-------------------------+---------------------------+
| Title | First name | Last name | Date of birth | National Insurance number | Payroll reference | Earnings in contribution period | Worker contribution (£) | Employer contribution (£) |
+-------+------------+-----------+---------------+---------------------------+-------------------+---------------------------------+-------------------------+---------------------------+
| MS    | JOE        | BLOGGS    |               |                           | BLOG              | 72.52                           | 0.00                    | 0.00                      |
| MS    | JOAN       | SMITH     |               | AB359288C                 | SMIT              | 171.50                          | 6.86                    | 5.15                      |
+-------+------------+-----------+---------------+---------------------------+-------------------+---------------------------------+-------------------------+---------------------------+
```

It seems all right

**Comment by davb06** [comment, marked as answer]
Solved! This is not a Miller issue, but a Windows batch file issue. Is a trap for the unwary.

Thanks @aborruso for your answer. This got me testing further. I found that the command - if run directly in the command prompt worked as you found.

I missed the fact that the issue arose in a batch file. The issue here is that a UK pound sign £ is not an ascii character, so is processed differently by the command prompt and Windows GUI programs like Notepad. It is fixed by changing the code page in the batch file to 650001 for unicode.

This idea was discussed at [Change the active console Code Page](https://ss64.com/nt/chcp.html), which explains that the default code page is determined by the Windows Locale.

For example, put this code at the start of your batch file:

```
:: Change the code page to Unicode/65001 before using non-ascii characters. See for example https://ss64.com/nt/chcp.html
chcp 65001
```
